src/psyclone/transformations.py

```diff
@@ -313,12 +313,161 @@ def apply(self, node, options=None):
         return schedule, keep
 
 
+class OMPTaskloopTrans(ParallelLoopTrans):
+
+    '''
+    Adds an OpenMP taskloop directive to a loop.
+
+    TODO: #1364 Taskloops do not yet support reduction clauses.
+
+    For example:
+
+    >>> from pysclone.parse.algorithm import parse
+    >>> from psyclone.psyGen import PSyFactory
+    >>> api = "gocean1.0"
+    >>> filename = "nemolite2d_alg.f90"
+    >>> ast, invokeInfo = parse(filename, api=api, invoke_name="invoke")
+    >>> psy = PSyFactory(api).create(invokeInfo)
+    >>>
+    >>> from psyclone.transformations import OMPParallelTrans, OMPSingleTrans
+    >>> from psyclone.transformations import OMPTaskloopTrans
+    >>> singletrans = OMPSingleTrans()
+    >>> paralleltrans = OMPParallelTrans()
+    >>> tasklooptrans = OMPTaskloopTrans()
+    >>>
+    >>> schedule = psy.invokes.get('invoke_0').schedule
+    >>> schedule.view()
+    >>>
+    # Apply the OpenMP Taskloop transformation to *every* loop
+    # in the schedule.
+    # FIXME: This ignores loop dependencies
+    >>> for child in schedule.children:
+    >>>     tasklooptrans.apply(child)
+    >>> # Enclose all of these loops within a single OpenMP
+    >>> # SINGLE region
+    >>> singletrans.apply(schedule.children)
+    >>> # Enclose all of these loops within a single OpenMP
+    >>> # PARALLEL region
+    >>> paralleltrans.apply(schedule.children)
+    >>> schedule.view()
+
+    '''
+    def __init__(self, grainsize=None, num_tasks=None):
+        self._grainsize = None
+        self._num_tasks = None
+        if grainsize is not None:
+            self.omp_grainsize = grainsize
+        if num_tasks is not None:
+            self.omp_num_tasks = num_tasks
+
+        super(OMPTaskloopTrans, self).__init__()
+
+    def __str__(self):
+        return "Adds an 'OpenMP TASKLOOP' directive to a loop"
+
+    @property
+    def omp_grainsize(self):
+        ''' Returns the grainsize that will be specified by
```

Comment:
Both getters need the ":returns:" and ":rtype:" docstring. If the ":returns:" docstring turns out to be the same as the method description (it often happens in getters) you can skip the method description.

Reply:
Done. I kept the method description to explain that this has a default value.